packages/sit-onyx/src/components/OnyxSelectInput/OnyxSelectInput.vue

```diff
@@ -52,83 +38,121 @@ const modelValueCount = computed(() => {
  * On multi select, it is a summary or a preview of the options.
  */
 const selectionText = computed<string>(() => {
-  if (Array.isArray(props.modelValue)) {
-    const numberOfSelections = props.modelValue.length;
+  if (Array.isArray(props.selection)) {
+    const numberOfSelections = props.selection.length;
     if (!numberOfSelections) return "";
-    if (numberOfSelections === 1) return props.modelValue[0].label;
+    if (numberOfSelections === 1) return props.selection[0].label;
 
-    switch (multipleTextMode.value) {
+    switch (props.textMode) {
       case "preview":
-        return props.modelValue.map(({ label }) => label).join(", ");
+        return props.selection.map(({ label }) => label).join(", ");
       case "summary":
       default:
         return t.value("selections.currentSelection", { n: numberOfSelections });
     }
   }
 
-  return props.modelValue?.label ?? "";
+  return props.selection?.label ?? "";
+});
+
+const rootAttrs = computed(
+  () =>
+    ({ class: attrs["class"], style: attrs["style"] }) as Pick<
+      HtmlHTMLAttributes,
+      "class" | "style"
+    >,
+);
+
+const inputAttrs = computed<Omit<HtmlHTMLAttributes, "class" | "style">>(() => {
+  const rest = { ...attrs };
+  delete rest.class;
+  delete rest.style;
+  return rest;
 });
```

Comment:
I think we also use this in other components, should we maybe create a composable for this?

Reply:
done :)